whisper/shhapi/api.go

```diff
@@ -368,60 +368,21 @@ func (api *PublicWhisperAPI) Post(args PostArgs) error {
 
 type PostArgs struct {
 	TTL      uint32              `json:"ttl"`
-	From     rpc.HexBytes        `json:"from"`
-	To       rpc.HexBytes        `json:"to"`
+	From     string              `json:"from"`
+	To       string              `json:"to"`
```

Comment:
I don't understand this change. RPC arguments need to be hex-encoded. You still do the hex decoding
but any invalid hex will silently post with empty From or To because common.FromHex returns nil on error.
Please keep the hex bytes. 

We have a new package, common/hexutil, since #3355. You can use it to get error-checked decoding
of uint64. This could be used for the filter ID, unless you specifically want them to be JSON numbers instead
of hex strings.